salt/minion.py

```diff
@@ -2196,11 +2196,14 @@ def pillar_refresh(self, force_refresh=False):
                     self.opts['saltenv'],
                     pillarenv=self.opts.get('pillarenv'),
                 ).compile_pillar()
+                if notify:
+                    evt = salt.utils.event.get_event('minion', opts=self.opts, listen=False)
+                    evt.fire_event({'complete': True}, tag='/salt/minion/minion_pillar_complete')
```

Comment:
I would move that thing into a constant. Since there is currently no nice home for this, I would keep it in this file, something like:

```python
PILLAR_COMPLETE = '/salt/minion/minion_pillar_complete'
```

And then I would [reuse it here](https://github.com/saltstack/salt/pull/50168/files#diff-a72c3e04b3df0479b574ff06020607efR1055).

Reply:
I changed the code accordingly.

Reply:
Okay, the last change went not well

```
Traceback (most recent call last):
  File "./tests/runtests.py", line 61, in <module>
    from tests.integration import TestDaemon  # pylint: disable=W0403
  File "/tmp/kitchen/testing/salt/tests/integration/__init__.py", line 36, in <module>
    from tests.support.case import ShellTestCase
  File "/tmp/kitchen/testing/salt/tests/support/case.py", line 36, in <module>
    from tests.support.mixins import AdaptedConfigurationTestCaseMixin, SaltClientTestCaseMixin
  File "/tmp/kitchen/testing/salt/tests/support/mixins.py", line 34, in <module>
    import salt.utils.event
  File "/tmp/kitchen/testing/salt/utils/event.py", line 79, in <module>
    import salt.payload
  File "/tmp/kitchen/testing/salt/payload.py", line 17, in <module>
    import salt.crypt
  File "/tmp/kitchen/testing/salt/crypt.py", line 64, in <module>
    import salt.utils.crypt
  File "/tmp/kitchen/testing/salt/utils/crypt.py", line 16, in <module>
    import salt.loader
  File "/tmp/kitchen/testing/salt/loader.py", line 39, in <module>
    from salt.minion import MinionEvents
  File "/tmp/kitchen/testing/salt/minion.py", line 66, in <module>
    import salt.client
  File "/tmp/kitchen/testing/salt/client/__init__.py", line 31, in <module>
    import salt.cache
  File "/tmp/kitchen/testing/salt/cache/__init__.py", line 16, in <module>
    from salt.payload import Serial
ImportError: cannot import name Serial
```
Is this due to a circular dependency?

Reply:
Moved the event constants into an own module `salt.events`, similar to  `salt.defaults`.

Reply:
@isbm, how to we proceed? After moving the constants into a dedicated module it works again.

Reply:
Not sure it should be a separate package though. If that so, I would rather put that thing to `salt/defaults/events.py` then. This would be a good home for this sort of things. Then we could gather those hard-coded tags all over the place and refer them there too.

@terminalmage any suggestions/objections?

Reply:
@isbm, I changed the PR accordingly.

Reply:
@isbm, how can we move this forward?